src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,207 @@
+        return newTask;
+
+    }
+
+        //Using format given by textbook dd-mm-yyyy HHmm example :"02/12/2019 1800"
+    /**
+     * Convert the LocalDateTime based on String formatted input
+     *
+     * @param input the input has to be format of dd-mm-yyyy HHmm, example : "02/12/2019 1800"
+     * @return The "translated" datetime based on input
+     */
+    public LocalDateTime convertDateTime(String input) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
+        LocalDateTime dateTime = LocalDateTime.parse(input, formatter);
+
+        return dateTime;
+    }
+
+
+    /**
+     * ProcessDeadline based on the userinput, slicing and categorise with text, dueDate
+     * then create and return a deadline task
+     *
+     * @return Deadline Task
+     */
+    public Task processDeadline() {
+        if(inputArray.length <= 2) {
+            return null;
+        }
+
+        String dueDate = "";
+        String extractedTask = String.join(" ", Arrays.copyOfRange(inputArray, 1, inputArray.length));
+
+        for(int i = 0; i < inputArray.length; i++) {
+            if(inputArray[i].equals("/by")) {
+                dueDate = String.join(" ", Arrays.copyOfRange(inputArray, i+1, inputArray.length));
+                extractedTask = String.join(" ", Arrays.copyOfRange(inputArray, 1, i));
+
+                break;
+            }
+        }
+
+        Task newTask = new Deadline(extractedTask,convertDateTime(dueDate));
+        return newTask;
+    }
+
+    /**
+     * ProcessToDo based on the userinput, slicing, save and create todo task
+     *
+     * @return Todo Task
+     */
+    public Task processToDo() {
+        if(inputArray.length <= 1) {
+            return null;
+        }
+
+        String extractedTask = String.join(" ", Arrays.copyOfRange(inputArray, 1, inputArray.length));
+        Task newTask = new ToDo(extractedTask);
+
+        return newTask;
+    }
+
+    public String getExtracted() {
+
+        if(inputArray.length <= 1) {
+            return "";
+        }
+
+        String extractedWord = String.join(" ", Arrays.copyOfRange(inputArray, 1, inputArray.length));
+
+        return extractedWord;
+    }
+
+
+
+    /**
+     * Return an Integer Index to be marked based on String input, and -1 if invalid
+     *
+     * @return Integer Index to be marked
+     */
+    public int processMarkIndex() {
+        try {
+            //from string to index
+            int index = Integer.parseInt(getIndex());
+            return index;
+        }catch(Exception e) {
+            return -1;
+        }
+    }
+    /**
+     * Return an Integer Index to be unmarked based on String input, and -1 if invalid
+     *
+     * @return Integer Index to be unmarked
+     */
+    public int processUnmarkIndex() {
+        try {
+            int index = Integer.parseInt(getIndex());
+            return index;
+        }catch(Exception e) {
```

Comment:
Yup you probably understand by this point so yeah I will stop pointing it out haha